### PR TITLE
Remove some raw SQL strings

### PR DESCRIPTION
**Pull request by cconard96**

### CODE REVIEW

File: inc/alert.class.php [hunk 1]
```diff
@@ -210,79 +210,126 @@ public static function findAllToNotify($params = []) {
...
-                              AND `$ttable`.`items_id` IN ($fndgroup)
-                           OR `$ttable`.`itemtype` = 'User'
-                              AND `$ttable`.`items_id` = ".$_SESSION['glpiID']."
-                        )";
+         $targets_sql = [
+            'OR' => [
+               [
+                  'AND' => [
+                     "$ttable.itemtype" => 'Profile',
+                     "$ttable.items_id" => $_SESSION['glpiactiveprofile']['id'],
```
Comment (cedric-anne) [inline inc/alert.class.php:231]: Condition on `all_items` is missing.
```suggestion
                     'OR' => [
                         "$ttable.items_id" => $_SESSION['glpiactiveprofile']['id'],
                         "$ttable.all_items" => 1,
                     ],
```

File: inc/alert.class.php [hunk 2]
```diff
@@ -210,79 +210,126 @@ public static function findAllToNotify($params = []) {
...
-                     AND (`$table`.`date_end` IS NULL
-                           OR `$table`.`date_end` > '$today'
-                           OR `$table`.`date_end` = '$today'
-                     )
-                  AND `is_deleted`='0' AND `is_active`='1'
-                  $entity_sql";
-      $iterator = $DB->request($query);
-      if ($iterator->numrows() < 1) {
+      $criteria = [
+         'SELECT' => ["$table.*"],
```
Comment (cedric-anne) [inline inc/alert.class.php:271]: There was a `DISTINCT` keyword in previous code.
```suggestion
         'SELECT' => ["$table.*"],
         'DISTINCT' => true,
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by stonebuzz — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by stonebuzz — CHANGES_REQUESTED** [review]

**Review by stonebuzz — APPROVED** [review]

**Review by Rom1-B — APPROVED** [review]

File: inc/alert.class.php [hunk 3]
```diff
@@ -240,86 +240,137 @@ public static function findAllToNotify($params = [])
...
+                        $table  => 'id'
+                    ]
+                ]
+            ],
+            'WHERE' => [
+                [$login_show_hidden_sql],
+                [
+                    'OR' => [
+                        ["$table.date_start" => ['<', $today]],
+                        ["$table.date_start" => $today],
```
Comment (cedric-anne) [inline inc/alert.class.php:339]: Il would probably be a little faster.
```suggestion
                        ["$table.date_start" => ['<=', $today]],
```

File: inc/alert.class.php [hunk 4]
```diff
@@ -240,86 +240,137 @@ public static function findAllToNotify($params = [])
...
+                        ["$table.date_start" => null],
+                    ]
+                ],
+                [
+                    'OR' => [
+                        ["$table.date_end" => ['>', $today]],
+                        ["$table.date_end" => $today],
```
Comment (cedric-anne) [inline inc/alert.class.php:346]:
```suggestion
                        ["$table.date_end" => ['>=', $today]],
```

File: inc/alert.class.php [hunk 5]
```diff
@@ -240,86 +240,137 @@ public static function findAllToNotify($params = [])
...
+                ],
+                'is_deleted' => 0,
+                'is_active'  => 1,
+            ]
+        ];
+        if (!empty($targets_sql)) {
+            $criteria['INNER JOIN'][$ttable]['ON'][] = $targets_sql;
+        }
+        if (!empty($entity_sql)) {
+            $criteria['WHERE'][] = [$entity_sql];
```
Comment (cedric-anne) [inline inc/alert.class.php:358]: No need to encapsulate it, it is already an array.
```suggestion
            $criteria['WHERE'][] = $entity_sql;
```

File: inc/alert.class.php [hunk 6]
```diff
@@ -240,86 +240,137 @@ public static function findAllToNotify($params = [])
...
+        }
+        if (!empty($login_sql)) {
+            $criteria['WHERE'][0][] = $login_sql;
+        }
+        if (!empty($show_central_sql)) {
+            $criteria['WHERE'][0][] = $show_central_sql;
+        }
+        if (!empty($show_helpdesk_sql)) {
+            $criteria['WHERE'][0][] = $show_helpdesk_sql;
+        }
```
Comment (cedric-anne) [inline inc/alert.class.php:368]: All of these conditions are mandatory and can be placed at the `AND` root level.
```suggestion
        if (!empty($login_sql)) {
            $criteria['WHERE'][] = $login_sql;
        }
        if (!empty($show_central_sql)) {
            $criteria['WHERE'][] = $show_central_sql;
        }
        if (!empty($show_helpdesk_sql)) {
            $criteria['WHERE'][] = $show_helpdesk_sql;
        }
```

File: inc/alert.class.php [hunk 7]
```diff
@@ -240,86 +240,137 @@ public static function findAllToNotify($params = [])
...
+            'INNER JOIN' => [
+                $ttable => [
+                    'ON' => [
+                        $ttable => 'plugin_news_alerts_id',
+                        $table  => 'id'
+                    ]
+                ]
+            ],
+            'WHERE' => [
+                [$login_show_hidden_sql],
```
Comment (cedric-anne) [inline inc/alert.class.php:335]: No need to encapsulate it (see comment related to other conditions).
```suggestion
                $login_show_hidden_sql,
```

**Comment by stonebuzz** [comment]
tested and approved

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**